### PR TITLE
Rest api - support allOf to merge

### LINKED ISSUE #360: Rest Swagger - support `allOf`
`allOf` should be supported according to  [Swagger schema object](http://swagger.io/specification/#schemaObject)

```json
"ParameterA": {
  "type": "object",
  "properties": {
    "error": {
      "$ref": "#/definitions/Object",
      "readOnly": true,
      "description": "sample"
    }
  },
  "allOf": [{
    "$ref": "#/definitions/ParameterB"
  }]
}
```

### PR DESCRIPTION
**Pull request by hellosnow**
Fix #360 , merge 'allOf' in templates

@vwxyzh @ansyral @chenkennt @superyyrrzz @qinezh

### CODE REVIEW

**Comment by vwxyzh** [comment]
👍

File: src/docfx.website.themes/default/RestApi.common.js [hunk 1]
```diff
@@ -19,6 +19,7 @@ exports.transform = function (model) {
             child.footer = child.footer || ''; // set to empty incase mustache looks up
 
             formatExample(child.responses);
+            resolveAllOf(model.children[i]);
```
Comment (qinezh) [inline src/docfx.website.themes/default/RestApi.common.js]: replace with `child` for consistency?

File: src/docfx.website.themes/default/RestApi.common.js [hunk 2]
```diff
@@ -79,6 +80,53 @@ exports.transform = function (model) {
...
+        else if (typeof obj === "object") {
+            for (var key in obj) {
+                if (obj.hasOwnProperty(key)) {
+                    if (key === "allOf" && Array.isArray(obj[key])) {
+                        // find 'allOf' array and process
+                        processAllOfArray(obj[key], obj);
+                        // set 'allOf' to undefined
+                        obj[key] = undefined;
+                    }
+                    else {
```
Comment (qinezh) [inline src/docfx.website.themes/default/RestApi.common.js]: minor: code style...

```javascript
if (...) {
  ...
} else {
  ...
}
```

File: src/docfx.website.themes/default/RestApi.common.js [hunk 3]
```diff
@@ -79,6 +80,53 @@ exports.transform = function (model) {
...
+            }
+        }
+        else if (typeof obj === "object") {
+            for (var key in obj) {
+                if (obj.hasOwnProperty(key)) {
+                    if (key === "allOf" && Array.isArray(obj[key])) {
+                        // find 'allOf' array and process
+                        processAllOfArray(obj[key], obj);
+                        // set 'allOf' to undefined
+                        obj[key] = undefined;
```
Comment (qinezh) [inline src/docfx.website.themes/default/RestApi.common.js]: how about removing "allOf" other than setting it to undefined?

**Comment by qinezh** [comment]
>         return obj;

No need to return obj? Since you already updated it in the function.

---
Refers to: src/docfx.website.themes/default/RestApi.common.js:146 in 842dd50. [](commit_id = 842dd50b40724509647566c4fa76583c37293813, deletion_comment = False)

**Comment by qinezh** [comment]
:shipit:

**Pull request merged**